### PR TITLE
remove soft fork logic at height 2300000

### PR DESCRIPTION
**Pull request by arvidn**
The whole chain can be validated with the post-soft-fork rules. Remove testing of pre-soft-fork rules.

This includes:

* removing the consensus constant `SOFT_FORK_HEIGHT`
* removing the `height` parameter from `run_generator()` and `get_name_puzzle_conditions()`, as it no longer depends on the block height (along with a whole bunch of passing around block height)
* simplify logic in `get_name_puzzle_conditions()` to always pass in the stricter flags (`COND_CANON_INTS` and `NO_NEG_DIV`)
* removing the `softfork_height` test fixture, which removes a bunch of test runs

There were also a few new member functions I had to add to the mocked peer and server objects in `test_full_sync.py` to make it pass.

### CODE REVIEW

File: chia/full_node/mempool_check_conditions.py [hunk 1]
```diff
@@ -25,31 +24,25 @@ def unwrap(x: Optional[uint32]) -> uint32:
...
-    # But otherwise, height must be specified to know which rules to apply
-    assert mempool_mode or height is not None
-
     # mempool mode also has these rules apply
     assert (MEMPOOL_MODE & COND_CANON_INTS) != 0
     assert (MEMPOOL_MODE & NO_NEG_DIV) != 0
 
     if mempool_mode:
         flags = MEMPOOL_MODE
-    elif unwrap(height) >= DEFAULT_CONSTANTS.SOFT_FORK_HEIGHT:
```
Comment (arvidn) [inline chia/full_node/mempool_check_conditions.py:46]: this is the root of this patch. The behavior of running the generator no longer depends on the block height

File: tools/run_block.py [hunk 2]
```diff
@@ -96,20 +95,10 @@ def npc_to_dict(npc: NPC):
...
-        # leading zeros)
-        # the division operator may not be used with negative operands
-        flags = COND_CANON_INTS | NO_NEG_DIV
-    else:
-        flags = 0
+def run_generator(block_generator: BlockGenerator, constants: ConsensusConstants, max_cost: int) -> List[CAT]:
 
     args = create_generator_args(block_generator.generator_refs).first()
-    _, block_result = block_generator.program.run_with_cost(max_cost, flags, args)
+    _, block_result = block_generator.program.run_with_cost(max_cost, 0, args)
```
Comment (arvidn) [inline tools/run_block.py:101]: This needs some investigation

Comment (arvidn) [reply tools/run_block.py:101]: `SerializedProgram.run_with_cost()` does not have a `flags` parameter, so this `flags` is interpreted as a parameter to the clvm program.

**Comment by arvidn** [comment]
both the `test_full_sync.py` and the `analyze_chain.py --mempool-mode` tests passed.

```
$ python tools/test_full_sync.py run ~/.chia/mainnet/db/blockchain_v2_mainnet.sqlite
PID: 2713322
Chia directory /tmp/tmpi9wfofj_
Can't find private CA, creating a new one in /tmp/tmpi9wfofj_ to generate TLS certificates

To see your keys, run 'chia keys show --show-mnemonic-seed'

height 100000 0.03 s/block
height 200000 0.02 s/block
height 300000 0.03 s/block
height 400000 0.03 s/block
height 500000 0.04 s/block
height 600000 0.03 s/block
height 700000 0.04 s/block
height 800000 0.05 s/block
height 900000 0.05 s/block
height 1000000 0.05 s/block
height 1100000 0.04 s/block
height 1200000 0.04 s/block
height 1300000 0.05 s/block
height 1400000 0.04 s/block
height 1500000 0.03 s/block
height 1600000 0.09 s/block
height 1700000 0.08 s/block
height 1800000 0.10 s/block
height 1900000 0.09 s/block
height 2000000 0.05 s/block
height 2100000 0.04 s/block
height 2200000 0.04 s/block
height 2300000 0.04 s/block
height 2312224 0.04 s/block   closing full node
```

```
./analyze-chain.py --mempool-mode ~/.chia/mainnet/db/blockchain_v2_mainnet.sqlite
```

**Review by jack60612 — DISMISSED** [review]
Looks good, i took a sec to realize what the `test_full_sync.py` changes are.

**Review by jack60612 — APPROVED** [review]
lGTM, gh says no changes besides rebase

**Review by trepca — APPROVED** [review]

**Comment by wjblanke** [comment]
nice to see we can do this

**Pull request merged**